### ISSUE TITLE
Rust support

### ISSUE THREAD

**Issue by FranklinYu** [github/semantic#130]
Any interest in Rust support? Should be easier than dynamic languages.

**Comment by robrix** [comment]
Thanks for the question! We aren’t currently focusing on it, but I’d be thrilled to review any work done to support it. If you’re interested in tackling it, #126 gives a brief outline of the process; note also that https://github.com/tree-sitter/tree-sitter-rust exists (tho I don’t know off the top of my head how complete it is).

cc @shaziyabandukia with whom I was literally just now talking about this 😁
cc @maxbrunsfeld for deets re: `tree-sitter-rust`’s completeness

**Comment by maxbrunsfeld** [comment]
Yeah, `tree-sitter-rust` is pretty complete. We've been using it in Atom for syntax highlighting, and it handles all the Rust code I've run it on. We recently got some community contributions that brought it up to speed with the newest language features.

I'd personally love it if semantic supported Rust, because Tree-sitter itself uses it quite a bit!

**Comment by p-alik** [comment]
I did some work to solve the issue yet https://github.com/p-alik/semantic/blob/semantic-rust/semantic-rust/README.md

**Comment by patrickt** [comment]
Now that the `tree-sitter-rust` Haskell package exists, next step is to create a `semantic-rust` package like the other ones we have in the semantic repo.

**Comment by p-alik** [comment]
But `tree-sitter-rust` is still not available on [hackage](https://hackage.haskell.org/packages/search?terms=tree-sitter-rust).  Consequently `cabal` can't compile `semantic-rust` yet.

**Comment by patrickt** [comment]
@p-alik Use the `cabal.project` feature to pin to the git repo directly.

**Comment by TheButlah** [comment]
Is Rust supported now? I noticed that #551 was merged

**Comment by p-alik** [comment]
> Is Rust supported now? I noticed that #551 was merged

Possible this [note in #572](https://github.com/github/semantic/issues/572#issuecomment-644252716) answers your question.
